server/src/main/java/org/elasticsearch/common/Strings.java

```diff
@@ -627,6 +627,45 @@ public static void collectionToDelimitedString(Iterable<?> coll, String delim, S
         }
     }
 
+    /**
+     * Converts a collection of items to a string like {@link #collectionToDelimitedString(Iterable, String, String, String, StringBuilder)}
+     * except that it stops if the string gets too long and just indicates how many items were omitted.
+     *
+     * @param coll        the collection of items to display
+     * @param delim       the delimiter to write between the items (usually {@code ","})
+     * @param prefix      a string to write before each item (usually {@code ""} or {@code "["})
+     * @param suffix      a string to write after each item (usually {@code ""} or {@code "]"})
+     * @param appendLimit if this many characters have been appended to the string and there are still items to display then the remaining
+     *                    items are omitted
+     */
+    public static void collectionToDelimitedStringWithLimit(
+        Iterable<?> coll,
+        String delim,
+        String prefix,
+        String suffix,
+        int appendLimit,
+        StringBuilder sb
+    ) {
+        final Iterator<?> it = coll.iterator();
+        final long lengthLimit = sb.length() + appendLimit; // long to avoid overflow
+        int count = 0;
+        while (it.hasNext()) {
+            sb.append(prefix).append(it.next()).append(suffix);
+            count += 1;
```

Comment:
I think`count++` would be more idiomatic rather than `count+=1`

Reply:
Idiom is a pretty subjective area, there isn't really a scale on which something is "more idiomatic" than something else. I think we should optimise for the reader and make assignments look like assignments wherever possible.